File: m3-h2/src/main/java/ru/yegorpilipenko/otus/spring/m3h2/service/GenreServiceImpl.java
package ru.yegorpilipenko.otus.spring.m3h2.service;

import lombok.RequiredArgsConstructor;
import org.springframework.security.access.annotation.Secured;
import org.springframework.stereotype.Service;
import ru.yegorpilipenko.otus.spring.m3h2.model.Genre;
import ru.yegorpilipenko.otus.spring.m3h2.repository.BookRepository;

import java.util.Collection;

import static ru.yegorpilipenko.otus.spring.m3h2.model.User.Role.ADMIN;
import static ru.yegorpilipenko.otus.spring.m3h2.model.User.Role.USER;

@Service
@Secured({ADMIN, USER})
@RequiredArgsConstructor
public final class GenreServiceImpl implements GenreService {

    private final BookRepository bookRepository;

    @Override
    public Collection<Genre> getAll() {
        return bookRepository.findAllGenres();
    }

}

File: m1-h5/src/main/kotlin/ru/yegorpilipenko/otus/spring/m1h5/service/GenreServiceImpl.kt
package ru.yegorpilipenko.otus.spring.m1h5.service

import org.springframework.stereotype.Service
import ru.yegorpilipenko.otus.spring.m1h5.dao.GenreDao
import ru.yegorpilipenko.otus.spring.m1h5.model.Genre

@Service
class GenreServiceImpl(val genreDao: GenreDao): GenreService {

    override fun findAll(): Collection<Genre> = genreDao.findAll()

    override fun findById(id: Long): Genre? = genreDao.findById(id)

    override fun save(genre: Genre): Genre = genreDao.save(genre)

    override fun removeById(id: Long): Boolean = genreDao.removeById(id)

}
